### PR TITLE
Add metrics collection for opening issueish in pane or browser

### PR DESCRIPTION
**Pull request by kuychaco**
### Description of the Change

There are currently six ways of viewing details for an issueish from our package (at least as far as I can tell... please let me know if you can think of another!). I have instrumented each of them and added a corresponding test. 

Added the following events and metadata:
* clicking an issueish link in an IssueishDetailView opens another IssueishDetailView in a new tab
`'open-issueish-in-pane', {from: 'issueish-link', target: 'new-tab'}`
* cmd-clicking an issueish link in an IssueishDetailView opens another IssueishDetailView in the current tab
`'open-issueish-in-pane', {from: 'issueish-link', target: 'current-tab'}`
* clicking an issueish item in the IssueishListView opens an IssueishDetailView in a new tab
`'open-issueish-in-pane', {from: 'issueish-list'}`
* pasting an issueish link in the OpenIssueishDialog opens an IssueishDetailView in a new tab
`'open-issueish-in-pane', {from: 'dialog'}`
* shift-clicking an issueish link in an IssueishDetailView opens the issueish page on dotcom
`'open-issueish-in-browser', {from: 'issueish-link'}`
* clicking the issueish reference in the header of the IssueishDetailView opens the corresponding issueish page on dotcom
`'open-issueish-in-browser', {from: 'issueish-header'}`

In the process I also caught a bug -- cmd-clicking an issueish link used to throw the following error:
<details>
<summary>Uncaught TypeError: _issueishDetailItem2.default.opener is not a function</summary>
Thrown From: [github](https://github.com/atom/github) package 0.19.0
At /Users/kuychaco/.atom/packages/github/lib/views/issueish-link.js:70
TypeError: _issueishDetailItem2.default.opener is not a function
    at openInNewTab (/packages/github/lib/views/issueish-link.js:70:37)
    at openIssueishLinkInNewTab (/packages/github/lib/views/issueish-link.js:40:5)
    at Object.handleClickEvent (/packages/github/lib/views/issueish-link.js:29:5)
    at BareGithubDotcomMarkdown.handleClick (/packages/github/lib/views/github-dotcom-markdown.js:101:18)
</details>

### Alternate Designs

Rather than having separate events for viewing the issueish in a pane or browser, I considered using only one event type of `open-issueish` and adding target pane/browser information to the metadata. I decided to keep them separate because there is a clear distinction between the intention as well as the impact of the two types of events, namely that `open-issueish-in-pane` keeps you in Atom whereas `open-issueish-in-browser` takes you out of Atom. 

### Benefits

These changes will shed light on behavior when viewing PR and issue information, potentially inform decisions around product and project work, as well as help us assess the impact of future changes we make. 

Here are some questions we might get insight into from collecting these metrics:
* How frequently are people leaving Atom to view information in the browser?
  * How does this change as we add more GitHub information and features into Atom?
* What entry points are being used to view issueish information? 
* Are people using the open issue dialog to specify a PR/issue to view? Currently you can only access this using a command `github:open-issue-or-pull-request`
* Are there discoverability issues for any of the entry points listed above? What is the impact of taking steps to improve discoverability?
* What are the usage statistics for the issue list view in the GitHub tab?
* What are the usage statistics for viewing issues/PRs that are referenced as part of an issue/PR that is being viewed?

### Possible Drawbacks

None that I can think of...

### Applicable Issues

#1591

### CODE REVIEW

File: lib/controllers/issueish-searches-controller.js [hunk 1]
```diff
@@ -93,7 +94,9 @@ export default class IssueishSearchesController extends React.Component {
         this.props.workingDirectory,
       ),
       {pending: true, searchAllPanes: true},
-    );
+    ).then(() => {
```
Comment (annthurium) [inline lib/controllers/issueish-searches-controller.js]: Hmm...maybe the reporter should have an async helper for recording async events. 🤔

File: test/views/github-dotcom-markdown.test.js [hunk 2]
```diff
@@ -84,31 +87,118 @@ describe('GithubDotcomMarkdown', function() {
...
     atom.commands.dispatch(link, 'github:open-link-in-this-tab');
-    assert.isTrue(switchToIssueish.calledWith('aaa', 'bbb', 123));
+    assert.isTrue(switchToIssueishStub.calledWith('aaa', 'bbb', 123));
 
     atom.commands.dispatch(link, 'github:open-link-in-browser');
-    assert.isTrue(openLinkInBrowser.calledWith(href));
+    assert.isTrue(openLinkInBrowserStub.calledWith(href));
+  });
+
+  describe('opening issueish links', function() {
```
Comment (annthurium) [inline test/views/github-dotcom-markdown.test.js]: would it make sense to add a unit test here to assert that if the link opening fails, we don't record an event?

Comment (kuychaco) [reply test/views/github-dotcom-markdown.test.js]: done! 👍

File: lib/controllers/issueish-searches-controller.js (hunk 1 above)
Comment (kuychaco) [reply lib/controllers/issueish-searches-controller.js]: Possibly! Personally, I think using `addEvent` like this is just fine -- it's clear and readable. That said, definitely open to trying it out and seeing how it feels / if there are additional benefits to using an async helper.

**Pull request merged**